provider/resource.go

```diff
@@ -275,7 +275,7 @@ func (r *resource) Read(f *filter.Filter) error {
 	// so we have to do it manually
 	// it's not all of them though
 	for _, t := range f.Tags {
-		if v, ok := r.data.GetOk(fmt.Sprintf("%s.%s", r.Provider().TagKey(), t.Name)); ok && v.(string) != t.Value {
+		if v, ok := r.data.GetOk(fmt.Sprintf("%s.%s", r.Provider().TagKey(), t.Name)); !ok || v.(string) != t.Value {
```

Comment:
Are you sure about this?

The `ok` is validating that the value exists, and then it validates that has the right value, by changing it if does not exists it will not be of type `string` and potentially fail the conversion. Also if it does not exists it makes no sense to compare it to `t.Value`.

So IDK if this is fixing anything or potentially adding a bug.

Reply:
You are right, one condition is missing. I should be something like `!ok || (ok && v.(string) != t.Value)`

Reply:
You are right, one condition is missing. I should be something like !ok || (ok && v.(string) != t.Value)

Reply:
Correct me if I'm wrong `!ok` ensure to skip/return if the key 'tags' doesn't exist. if it doesn't exist, the first condition will be `!ok -> true` and the second condition will not be tested.
And only when the key "tags" exist, we will test the second condition ?



`!ok` || v.(string) != t.Value

Reply:
Ok yes I read it too quick haha.